File: 1566-check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence/1566-check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence.cpp
class Solution {
public:
    int isPrefixOfWord(string sentence, string searchWord) {
        stringstream s(sentence);
        string word;
        int idx=1;
        while(s>>word){
            if(word.find(searchWord)==0){
                return idx;
            }
            idx++;
        }
        return -1;
    }
};

File: 1566-check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence/1566-check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence.py
class Solution(object):
    def isPrefixOfWord(self, sentence, searchWord):
        """
        :type sentence: str
        :type searchWord: str
        :rtype: int
        """
        w=sentence.split()
        words=list(enumerate(w))
        for i in words:
            if i[1].startswith(searchWord):
                return i[0]+1
        
        return -1